packages/flutter_driver/lib/src/driver/timeline_summary.dart

```diff
@@ -225,7 +231,12 @@ class TimelineSummary {
         .reduce(math.max);
   }
 
-  SceneDisplayLagSummarizer _sceneDisplayLagSummarizer() => SceneDisplayLagSummarizer(_extractNamedEvents(kSceneDisplayLagEvent));
+  SceneDisplayLagSummarizer _sceneDisplayLagSummarizer() =>
+      SceneDisplayLagSummarizer(_extractNamedEvents(
+        kSceneDisplayLagEvent,
+        filter: (TimelineEvent event) =>
+            event.arguments.containsKey(kVsyncTransitionsMissed),
```

Comment:
They are only on the begin events so you can filter on phase == 'b' or 'B' (depending on whether it is async or sync). It would be faster than a hash lookup.

Reply:
I believe that @iskakaushik is deprecating the async events and transitioning to instant events in https://github.com/flutter/engine/pull/17918. So there would be soon no `phase`, and all `kSceneDisplayLagEvent` should have `kVsyncTransitionsMissed` key?

Reply:
Yes, he is converting to synchronous events, but that actually doubles down on this issue. With `AsyncEventComplete` it put the args on both the begin and end events. With synchronous events, the args only go on the begin event. Both techniques had both begin and end events in any case.

(Note that "instant event" is not an actual construct in the timeline, it is just a synchronous event with no instructions to surround with its auto-end mechanism so the begin/end happen back to back.)

So, after the change to a synchronous event it becomes even more important to only examine the begin events.

Reply:
https://github.com/flutter/flutter/blob/master/packages/flutter_driver/lib/src/driver/timeline.dart#L131

I'm leaning towards sorting the events based on `ts` here. I went down the rabbit hole of trying to figure out why the events on the same thread can get reordered, but i don't have any leads.

There might be an issue with the endless recorder. We've already identified a potential inconsistency and fixed it at https://dart-review.googlesource.com/c/sdk/+/144810. I am surprised that nobody else is seeing re-ordered events and that only the `SceneDisplayLag` is resulting in them.

WDYT?

Reply:
@flar : Oh, I just realized that flutter/engine#17918 is using `TRACE_EVENT1` instead of `TRACE_EVENT_INSTANT`.

@iskakaushik : maybe it's better to define `TRACE_EVENT_INSTANT1` and use it instead of `TRACE_EVENT1` so there would be only one event instead of a very close pair of begin/end events?

Reply:
Given how huge our event streams are getting I like the idea of minimizing the data in them. If the instant gets rid of another "event" in the stream, then all the more power to it.